engine/engine.go

```diff
@@ -41,7 +44,18 @@ func NewAuthEngine() *nutsGo.Engine {
 		Routes: func(router nutsGo.EchoRouter) {
 			// Mount the irma-app routes
 			routerWithAny := router.(EchoRouter)
-			irmaEchoHandler := echo.WrapHandler(authBackend.IrmaServer.HandlerFunc())
+
+			// The Irma router operates on the mount path and does not know about the prefix.
+			prefix := "/auth/irmaclient"
+			rewriteFunc := func(w http.ResponseWriter, r *http.Request) {
+				if strings.HasPrefix(r.URL.Path, prefix) {
+					// strip the prefix
+					r.URL.Path = strings.Split(r.URL.Path, prefix)[1]
+				}
+				authBackend.IrmaServer.HandlerFunc()(w, r)
+			}
+			// wrap the http handler in a echo handler
+			irmaEchoHandler := echo.WrapHandler(http.HandlerFunc(rewriteFunc))
 			routerWithAny.Any("/auth/irmaclient/*", irmaEchoHandler)
```

Comment:
Ahhh I see, nasty....

Is there an option to add a test? So when either Irma or Chi changes something to the relative routing, we would know easily?

Reply:
Added 👍